File: packages/js-sdk/src/regex/detectors/nz/ird.ts
// New Zealand IRD (Inland Revenue Department) Number detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { nzIrdChecksum } from '../../validators'

class NzIrdDetector extends RegexDetector {
  entityType = EntityType.NZ_IRD
  score = 0.85
  contextKeywords = ['ird', 'inland revenue', 'tax number', 'ird number']
  contextRequired = true

  // 8-9 digits with optional separators
  pattern = /\b\d{2,3}[- ]?\d{3}[- ]?\d{3}\b/g
  validator = nzIrdChecksum
}

registerRegion('nz', NzIrdDetector)

export { NzIrdDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/nz/ird.py
"""New Zealand IRD (Inland Revenue Department) Number detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import nz_ird_checksum


@register_region("nz")
class NzIrdDetector(RegexDetector):
    entity_type = EntityType.NZ_IRD
    score = 0.85
    context_keywords = [
        "ird", "inland revenue", "tax number", "ird number",
    ]
    context_required = True
    context_window = 50

    # 8-9 digits with optional separators
    pattern = re.compile(
        r"\b\d{2,3}[- ]?\d{3}[- ]?\d{3}\b"
    )
    validator = staticmethod(nz_ird_checksum)
